### DISCUSSION TITLE
Issue with Lint Rules Not Respecting Configuration in Subdirectory

### DISCUSSION THREAD

**Discussion by GOWxx** [biomejs/biome#2289] [Q&A]
Hello,

I've encountered a problem that I can't seem to solve on my own. Below is the structure of my project directory for reference:

```bash
.
├── biome.jsonc
├── code
│   ├── biome.jsonc
│   └── index.js
└── root.js
```

The issue I'm encountering is that the linting for `./code/index.js` does not adhere to the configurations set in its sibling `biome.jsonc` file, instead, it only responds to the configurations set in the root `./biome.jsonc` file. This behavior persists regardless of whether the `lint.enable` setting in the configuration is set to `true` or `false`. However, the configuration specified in the root `./biome.jsonc` file is effective.

I have tested two scenarios:
1. With `lint.enable` set to `false` in `./biome.jsonc`, and set to `true` in `./code/index.js`, the linting for `./code/index.js` does not work.
2. With `lint.enable` set to `true` in `./biome.jsonc`, and set to `false` in `./code/index.js`, linting still applies to `./code/index.js` as if the setting was not `false`.

Additionally, attempting to use `overrides` in `./code/index.js` has no effect.

It's worth noting that this behavior is consistent across both CLI and LSP, indicating that only the root directory configuration is recognized.

Could anyone please advise on how to resolve this issue so that the subdirectory's configuration is respected? 

I have provided a repository to replicate the issue, which can be found at: 
https://github.com/GOWxx/vscode-biome-nested-folder-bug-demo

Thank you in advance for your assistance!

**Comment by GOWxx** [comment]
**Update: I've made some progress on this issue.**

I managed to address the problem of executing commands in different directories using package scripts in combination with the `pnpm -r` command while utilizing the CLI.

Here are the steps I followed in each of the project folders:

1. Initialized the package with `pnpm init`.
2. Added Biome as a development dependency with the command `pnpm add --save-dev --save-exact @biomejs/biome`.
3. Initialized Biome with `pnpm biome init --jsonc`.

Then, I added the following script to each `package.json`:

```json
"lint": "pnpm biome lint ./"
```

By running `pnpm -r run lint` in the root directory, the package script `lint` is executed in every project that contains a `package.json` file.

Additionally, I utilized a `pnpm-workspace.yaml` configuration:

```yaml
packages:
- 'example'
```

With this setup, executing `pnpm -r run lint` will only run the script in the projects listed under `packages`.

**Comment by GOWxx** [comment]
However, I haven't been able to resolve the issue with the Visual Studio Code extension. My VSCode still follows the lint rules specified in the root directory's biome.jsonc, completely ignoring the configurations in subdirectories. How can I make the VSCode extension adhere to the nearest configuration file?

**Reply by ematipico** [reply]
https://biomejs.dev/reference/vscode/#my-biomejson-file-is-ignored-in-a-multi-root-workspace

**Reply by GOWxx** [reply]
Thank you, I found the relevant issues, PRs, and documentation content today. I will post the relevant links below to help others who encounter this issue.

- Related PRs
	- https://github.com/biomejs/biome/pull/2297
- Related issues
  - https://github.com/biomejs/biome/issues/1573
  - https://github.com/biomejs/biome/issues/1630
  - https://github.com/biomejs/biome/issues/2036
